### ISSUE TITLE
Allow one filename when using --stdin

### ISSUE THREAD

**Issue by glenjamin** [eslint/eslint#1950]
Editor-integrations can use a combination of `--stdin` and the current working directory to lint a file that has yet to be saved - but this means that any rule which relies on the filename doesn't work.

See https://github.com/roadhump/SublimeLinter-eslint/issues/28 for some discussion on this.

I'm proposing that:
* Passing multiple filenames alongside `--stdin` will raise an error
* Passing a single filename alongside `--stdin` will use that filename for linting

Currently filenames passed at the same time as `--stdin` are ignored.

Happy to submit a PR for this.

**Comment by nzakas** [comment]
I'm not sure about this. The whole point of piping text in is to not use a filename.

**Comment by glenjamin** [comment]
The particular scenario I'd like to support is linting a specific file before it is changed in the exact same way it would lint after save - this way I should never save an invalid file.

This is quite a specific scenario, but I think this is what most editor integrations would want to do.

**Comment by nzakas** [comment]
I'd need to hear from the people who've done the editor integrations before I can really consider this. This suggestion has the potential to be pretty confusing, so I'm inclined not to do it unless there's a pressing need.

**Comment by tikotzky** [comment]
Just bumped into this exact issue also while using https://github.com/roadhump/SublimeLinter-eslint

I'm creating a custom rule which gets triggered based on the filename. The reason I'm doing that is because we're in middle of converting our codebase from ES6 to ES6 and would like to have some rules only apply to ES6 files (for example `no-var`).

What I was trying to do was create a custom plugin that looked like...
```javascript
module.exports = {
	rules: {
		"no-var": function(context) {
			return {
				"VariableDeclaration": function (node) {
					// Only apply this rule if the filename ends in .es6
					if (node.kind === "var" && context.getFilename().slice(-4) === ".es6") {
						context.report(node, "Unexpected var, use let or const instead.");
					}
				}
			};
		}
	},
	rulesConfig: {
		"no-var": 2
	}
};
```

@nzakas perhaps instead of allowing you to pass filenames as you would when calling without `--stdin` there can be another option something like `--stdin-filename` or maybe change the current `--stdin` from being a boolean to allow you to pass a filename like `--stdin=test.es6`

**Comment by nzakas** [comment]
`--stdin-filename` seems like the best option.
